src/io/flutter/utils/MostlySilentOsProcessHandler.java

```diff
@@ -30,14 +33,43 @@
  * for more information.
  */
 public class MostlySilentOsProcessHandler extends OSProcessHandler {
+  /*
+  This determines whether a soft kill (SIGINT) is sent to the process on destroy, instead of the default SIGKILL. SIGKILL can't be routed
+  to remote processes spawned from the original one.
+   */
+  private boolean softKill;
+  private GeneralCommandLine commandLine;
+
   public MostlySilentOsProcessHandler(@NotNull GeneralCommandLine commandLine)
+    throws ExecutionException {
+    this(commandLine, false);
+  }
+
+  public MostlySilentOsProcessHandler(@NotNull GeneralCommandLine commandLine, boolean softKill)
     throws ExecutionException {
     super(commandLine);
+    this.softKill = softKill;
+    this.commandLine = commandLine;
   }
 
   @NotNull
   @Override
   protected BaseOutputReader.Options readerOptions() {
     return BaseOutputReader.Options.forMostlySilentProcess();
   }
+
+  @Override
+  protected void doDestroyProcess() {
+    final Process process = getProcess();
+    if (softKill && SystemInfo.isUnix && shouldDestroyProcessRecursively() && processCanBeKilledByOS(process)) {
+      final boolean result = UnixProcessManager.sendSigIntToProcessTree(process);
+      if (!result) {
+        FlutterInitializer.getAnalytics().sendEvent("process", "process kill failed: " + this.commandLine.getCommandLineString());
```

Comment:
don't need the full command line string which might identify the user, just the file name.